File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/command/solution/SimpleDriverCommand.java
// SimpleDriverCommand.java

public abstract class SimpleDriverCommand extends Command
{
	public SimpleDriverCommand(Driver aDriver)
	{
		setDriver(aDriver);
	}

	public abstract void execute();

	public abstract void undo();

	private void setDriver(Driver aDriver)
	{
		if (aDriver == null)
		{
			throw new DpException("void SimpleDriverCommand.setDriver(Driver aDriver) -- aDriver == null.");
		}

		myDriver = aDriver;
	}

	protected Driver getDriver()
	{
		return myDriver;
	}

	private Driver myDriver;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/solution/simple_driver_command.cpp
// simple_driver_command.cpp

#include "dp_exception.h"
class driver;
#include "simple_driver_command.h"

simple_driver_command::simple_driver_command(driver * a_driver)
{
	set_driver(a_driver);
}

void simple_driver_command::set_driver(driver * a_driver)
{
	if (a_driver == nullptr)
	{
		throw dp_exception("void simple_driver_command::set_driver(driver * a_driver) -- a_driver == nullptr.");
	}

	my_driver = a_driver;
}

driver * simple_driver_command::get_driver() const
{
	return my_driver;
}
